lib/vnet/app_resolver.go

```diff
@@ -159,21 +166,39 @@ func (r *TCPAppResolver) resolveTCPHandlerForCluster(
 		Limit:               1,
 	})
 	if err != nil {
-		// Don't return an error so we can try to find the app in different clusters or forward the request
-		// upstream.
+		// Don't return an unexpected error so we can try to find the app in different clusters or forward the
+		// request upstream.
 		slog.InfoContext(ctx, "Failed to list application servers.", "error", err)
-		return nil, false, nil
+		return nil, ErrNoTCPHandler
 	}
 	if len(resp.Resources) == 0 {
 		// Didn't find any matching app, forward the request upstream.
-		return nil, false, nil
+		return nil, ErrNoTCPHandler
 	}
 	app := resp.Resources[0].GetApp()
 	appHandler, err := r.newTCPAppHandler(ctx, profileName, leafClusterName, app)
 	if err != nil {
-		return nil, false, trace.Wrap(err)
+		return nil, trace.Wrap(err)
 	}
-	return appHandler, true, nil
+
+	var cidrRange string
+	vnetConfig, err := r.clusterConfigCache.getVnetConfig(ctx, profileName, leafClusterName)
+	switch {
+	case err == nil:
+		cidrRange = vnetConfig.GetSpec().GetIpv4CidrRange()
+		if cidrRange == "" {
+			cidrRange = defaultIPv4CIDRRange
+		}
+	case trace.IsNotFound(err) || trace.IsNotImplemented(err):
+		cidrRange = defaultIPv4CIDRRange
+	default:
+		return nil, trace.Wrap(err)
+	}
+
+	return &TCPHandlerSpec{
+		IPv4CIDRRange: cidrRange,
```

Comment:
We should log the CIDR range somewhere. I think it should make it slightly easier to debug a scenario where the default range is picked for whatever reason and it conflicts with other software on the device.

I see that there's "Running Teleport VNet" that already includes `ipv6_prefix`.

Reply:
can't really do it there since the ipv4 ranges are dynamic but it's already logged here https://github.com/gravitational/teleport/blob/380c137f797e48081f1aa5fe150f5e7c4e39206a/lib/vnet/osconfig_darwin.go#L46-L47

Reply:
to be fair though that log is in the root process which is currently written to `/var/log/vnet.log` https://github.com/gravitational/teleport/blob/530d231a221a3297be188ed01b07a2b26afd3e57/lib/vnet/setup_darwin.go#L117